microovn/ovn/commands.go

```diff
@@ -16,12 +16,14 @@ const defaultDBConnectWait = 30 //Default time to wait for connection to ovsdb
 const OvsdbConnected = "connected"
 const OvsdbRemoved = "removed"
 
-// ovsdbSpec is a helper structure for precise identification of ovsdb databases. A lot of
-// ovn/ovs commands take path to either database file, database socket or process control socket
-// along with the database name. This structure can be used for such cases.
-type ovsdbSpec struct {
-	Target string
-	Name   string
+// OvsdbSpec is a helper structure that encapsulates properties of an OVN/OVS database.
+type OvsdbSpec struct {
+	SocketPath   string // Path to a socket that the database has open
+	Name         string // Name of the database within the db file (e.g. "OVN_Northbound")
+	ShortName    string // Shorthand name for the database (e.g. "nb")
+	FriendlyName string // Human friendly name of the database ideal for logging purposes (e.g. "Northbound")
+	Schema       string // Path to a schema file for the database
+	IsCentral    bool   // Whether the database is used by OVN central services
```

Comment:
Adding a bool for this purpose feels redundant since we already have a notion of OvsdbType.

Can we record the OvsdbType instead and perhaps call a helper function which knows which of the types are used for central DBs?

Reply:
I guess that this is a matter of perspective. If we replaced `IsCentral` with something like `DbType`, that would hold `OvsdbType`, then we would still have same number of members in the `OvsdbSpec`, but now we'd also need a helper method that would decide whether an instance of `OvsdbSpec` is central or not.

In a sense, we already have a helper like this. It's the constructor for the `OvsdbSpec`. You pass `OvsdbType` to `NewOvsdbSpec` and it fills in the `IsCentral` member with true/false.

Reply:
Yea, I guess it is a matter of preference of using methods that understand the data structure rather than accessing the data directly from foreign functions/method.

Right now it does not make much difference so I'm fine either way, but if we somewhere down the road need to make other decisions on the back of the different types we might find it makes more sense to store the type.